src/cr/cube/matrix/measure.py

```diff
@@ -1545,21 +1545,27 @@ def _column_bases(self):
             calculated column bases (either 'effective' or unweighted counts) for
             the analysis.
         """
-        unweighted_blocks = self._second_order_measures.column_unweighted_bases.blocks
+        # unweighted_blocks = self._second_order_measures.column_unweighted_bases.blocks
+        weighted_blocks = self._second_order_measures.column_weighted_bases.blocks
         if self._second_order_measures.columns_squared_base.is_defined:
             squared_blocks = self._second_order_measures.column_squared_bases.blocks
             effective_blocks = [
                 [
-                    unweighted_blocks[0][0] ** 2 / squared_blocks[0][0],
-                    unweighted_blocks[0][1] ** 2 / squared_blocks[0][1],
+                    # unweighted_blocks[0][0] ** 2 / squared_blocks[0][0],
+                    # unweighted_blocks[0][1] ** 2 / squared_blocks[0][1],
+                    weighted_blocks[0][0] ** 2 / squared_blocks[0][0],
+                    weighted_blocks[0][1] ** 2 / squared_blocks[0][1],
                 ],
                 [
-                    unweighted_blocks[1][0] ** 2 / squared_blocks[1][0],
-                    unweighted_blocks[1][1] ** 2 / squared_blocks[1][1],
+                    # unweighted_blocks[1][0] ** 2 / squared_blocks[1][0],
+                    # unweighted_blocks[1][1] ** 2 / squared_blocks[1][1],
+                    weighted_blocks[1][0] ** 2 / squared_blocks[1][0],
+                    weighted_blocks[1][1] ** 2 / squared_blocks[1][1],
                 ],
             ]
             return effective_blocks
-        return unweighted_blocks
+        # return unweighted_blocks
+        return weighted_blocks
```

Comment:
I think the outer return here is what happens in the absence of weighting, or absence of w^2. In the first case (no weights) weighted == unweighted, which would be fine. In the second instance, where you might have weights but fail to supply w^2, unweighted would still be correct here. The bug was only in the effective calculation above.

Reply:
@malecki Can you re-check this? I think I figured it out finally.